deps/llvm.mk

```diff
@@ -94,16 +94,15 @@ LLVM_LDFLAGS += $(BOLT_LDFLAGS)
 LLVM_CMAKE += -DLLVM_TARGETS_TO_BUILD:STRING="$(LLVM_TARGETS)" -DCMAKE_BUILD_TYPE="$(LLVM_CMAKE_BUILDTYPE)"
 LLVM_CMAKE += -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD:STRING="$(LLVM_EXPERIMENTAL_TARGETS)"
 LLVM_CMAKE += -DLLVM_ENABLE_LIBXML2=OFF -DLLVM_HOST_TRIPLE="$(or $(XC_HOST),$(BUILD_MACHINE))"
-LLVM_CMAKE += -DLLVM_ENABLE_ZLIB=ON -DZLIB_LIBRARY="$(build_prefix)/lib"
+LLVM_CMAKE += -DLLVM_ENABLE_ZLIB=ON -DZLIB_ROOT="$(build_prefix)"
```

Comment:
I think we may want to set `LLVM_ENABLE_ZLIB=FORCE_ON`, which is the only way to make the check fail loudly:
https://github.com/llvm/llvm-project/blob/46425b8d0fac3c529aa4a716d19abd7032e452f3/llvm/cmake/config-ix.cmake#L143-L145.  With `LLVM_ENABLE_ZLIB=ON` cmake finds zlib, but the test at https://github.com/llvm/llvm-project/blob/46425b8d0fac3c529aa4a716d19abd7032e452f3/llvm/cmake/config-ix.cmake#L139-L141 is broken because `CMAKE_REQUIRED_LIBRARIES` lists a directory instead of a library, but `LLVM_ENABLE_ZLIB` is silently set to 0.